### DISCUSSION TITLE
x-bind broken/changed in AlpineJS > 3.7.1

### DISCUSSION THREAD

**Discussion by bep** [alpinejs/alpine#3290] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2942

<div type='discussions-op-text'>

<sup>Originally posted by **bep** May 19, 2022</sup>
See https://codepen.io/bepsays/pen/zYRweYj

I have some AlpineJS sites that broke when upgrading to 3.8. They have x-bind constructs similar to the below:

```js
<!DOCTYPE html>
<html>
<script defer src="https://unpkg.com/alpinejs@3.7.1/dist/cdn.min.js"></script>
<body x-data="foo">
  <style>
    .red {
      color: red;
    }
    .blue {
      color: blue;
    }
  </style>
  <button type="button" x-bind="myClass()" x-on:click="clickMe()">Click Me</button>
  <script>
    document.addEventListener('alpine:init', () => {
      Alpine.data('foo', () => ({
        counter: 0,
        myClass: function() {
          return {
            'class': this.counter % 2 == 0 ? 'blue' : 'red'
          }
        },
        clickMe() {
          this.counter++;
        },
      }))
    })
  </script>
</body>
</html>
```

In `3.7.1` this works as expected. In > 3.7.1, the above just fails silently. I have tried to read the change log to understand how to adapt ...</div>

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Comment by bep** [comment]
@joshhanley you are converting issues that also has a discussion thread into a discussion, effectively creating 2 discussion threads about the same.

**Reply by joshhanley** [reply]
@bep yep, some issues had a bunch of comments on them though that I didn't want to be lost, plus I figured some of those existing discussions are probably buried now, so felt it better just to convert into new ones.

**Comment by ekwoka** [comment]
I think this can be solved by changing the whole syntax to reflect how the docs do it:

https://alpinejs.dev/directives/bind#bind-directives

That's at least a good start.

Also "fails silently" means that it is not reactive? or doesn't even do it to start with?
